File: golang/cmd/transfer_domain/transfer_domain.go
// Usage: TOKEN=token go run cmd/create_domain/create_domain.go example.com 42 code
// where 42 is the contact id that will be used to register the domain and
// code is the authorization code do transfer the domain.
package main

import (
	"context"
	"fmt"
	"os"
	"strconv"

	"github.com/dnsimple/dnsimple-go/v5/dnsimple"
	"golang.org/x/oauth2"
)

func main() {
	oauthToken := os.Getenv("TOKEN")
	ts := oauth2.StaticTokenSource(&oauth2.Token{AccessToken: oauthToken})
	tc := oauth2.NewClient(context.Background(), ts)

	client := dnsimple.NewClient(tc)
	client.BaseURL = "https://api.sandbox.dnsimple.com"

	// get the current authenticated account (if you don't know who you are)
	whoamiResponse, err := client.Identity.Whoami(context.Background())
	if err != nil {
		fmt.Printf("Whoami() returned error: %v\n", err)
		os.Exit(1)
	}

	accountID := strconv.FormatInt(whoamiResponse.Data.Account.ID, 10)

	fmt.Printf("account id %v\n", accountID)

	name := os.Args[1]
	registrantID, _ := strconv.Atoi(os.Args[2])
	authCode := os.Args[3]

	transferRequest := &dnsimple.TransferDomainInput{RegistrantID: registrantID, AuthCode: authCode}

	fmt.Printf("Transferring domain %v\n", name)
	transferDomainResponse, err := client.Registrar.TransferDomain(context.Background(), accountID, name, transferRequest)
	if err != nil {
		fmt.Printf("TransferDomain() returned error: %v\n", err)
		os.Exit(1)
	}
	fmt.Printf("%+v\n", transferDomainResponse.Data)
}

File: csharp/Examples/TransferDomain.cs
using dnsimple;
using dnsimple.Services;

namespace DnsimpleExamples.Examples;

// Usage: TOKEN=your-token dotnet run -- transfer_domain example.com 42 code
// where 42 is the contact id that will be used as the registrant and code
// is the authorization code needed to transfer the domain.
public static class TransferDomain
{
    public static void Run(string[] args)
    {
        var token = Environment.GetEnvironmentVariable("TOKEN");
        if (string.IsNullOrEmpty(token))
        {
            Console.Error.WriteLine("The TOKEN environment variable is required");
            Environment.Exit(1);
        }

        if (args.Length < 3 || !long.TryParse(args[1], out var registrantId))
        {
            Console.Error.WriteLine("Usage: dotnet run -- transfer_domain <domain-name> <registrant-contact-id> <auth-code>");
            Environment.Exit(1);
            return;
        }

        var name = args[0];
        var authCode = args[2];

        var client = new Client();
        client.ChangeBaseUrlTo("https://api.sandbox.dnsimple.com");
        client.AddCredentials(new OAuth2Credentials(token));

        // get the current authenticated account (if you don't know who you are)
        var accountId = client.Identity.Whoami().Data.Account.Id;

        Console.WriteLine($"account id {accountId}");
        Console.WriteLine($"Transferring domain {name}");

        var transferInput = new DomainTransferInput { RegistrantId = registrantId, AuthCode = authCode };

        // transfer a domain name in from another registrar
        try
        {
            var transfer = client.Registrar.TransferDomain(accountId, name, transferInput).Data;
            Console.WriteLine($"Transfer: Id={transfer.Id}, DomainId={transfer.DomainId}, State={transfer.State}");
        }
        catch (DnsimpleException e)
        {
            Console.Error.WriteLine($"TransferDomain() returned error: {e.Message}");
            Environment.Exit(1);
        }
    }
}
